pkg/fs_algo/fs_algo/fs_perf_viz.py

```diff
@@ -0,0 +1,428 @@
+
+    with open(path_viz_config, 'r') as file:
+        viz_cfg = yaml.safe_load(file)
+
+    # Get features from the viz config file --------------------------
+    algos = viz_cfg.get('algos')
+    print('Visualizing data for the following RaFTS algorithms:')
+    print(algos)
+    print('')
+    metrics = viz_cfg.get('metrics')
+    print('And for the following evaluation metrics:')
+    print(metrics)
+    print('')
+
+    plot_types = viz_cfg.get('plot_types')
+    plot_types_dict = {k: v for d in plot_types for k, v in d.items()}
+    true_keys = [key for key, value in plot_types_dict.items() if value is True]
+    print('The following plots will be generated:')
+    print(true_keys)
+    print('')
+
+    # Get features from the pred config file --------------------------
+    path_pred_config = fsate.build_cfig_path(path_viz_config,viz_cfg.get('name_pred_config',None)) # currently, this gives the pred config path, not the attr config path
+    pred_cfg = yaml.safe_load(open(path_pred_config, 'r'))
+    path_attr_config = fsate.build_cfig_path(path_pred_config,pred_cfg.get('name_attr_config',None)) 
+
+    # Get features from the attr config file --------------------------
+    with open(path_attr_config, 'r') as file:
+        attr_cfg = yaml.safe_load(file)
+
+    datasets = list([x for x in attr_cfg['formulation_metadata'] if 'datasets' in x][0].values())[0] # Identify datasets of interest
+    dir_base = list([x for x in attr_cfg['file_io'] if 'dir_base' in x][0].values())[0]
+    dir_std_base = list([x for x in attr_cfg['file_io'] if 'dir_std_base' in x][0].values())[0]
+    dir_std_base = f'{dir_std_base}'.format(dir_base = dir_base)
+    # Options for getting ds_type from a config file:
+    # ds_type = viz_cfg.get('ds_type') # prediction config file IF VISUALIZING PREDICTIONS; attribute config file IF AND ONLY IF VISUALIZING ATTRIBUTES
+    # ds_type = list([x for x in attr_cfg['file_io'] if 'ds_type' in x][0].values())[0]
+    # ...but for plotting purposes, we want to use the prediction ds_type:
+    ds_type = 'prediction'
+    write_type = list([x for x in attr_cfg['file_io'] if 'write_type' in x][0].values())[0]
+
+    # Get features from the main config file --------------------------
+    # NOTE: This assumes that the main config file is just called [same prefix as all other config files]_config.yaml
+    prefix_viz = str(path_viz_config.name).split('_')[0]
+    prefix_attr = str(path_attr_config.name).split('_')[0]
+    if (prefix_viz != prefix_attr):
+        raise ValueError('The base config file (e.g. [dataset]_config.yaml) must be in the same direcotry and identifiable using the same prefix as the other config files (e.g. [dataset]_pred_config.yaml, [dataset]_attr_config.yaml, etc.)')
+    else:
+        prefix = prefix_viz
+
+    path_main_config = fsate.build_cfig_path(path_viz_config,f'{prefix_viz}_config.yaml')
```

Comment:
Note I am actually referring to the main config file here. The one called [dataset]_config.yaml. I did add a little bit of extra documentation here though the explain that I'm building the path to it using the other two config files that this script already reads in in order to avoid requiring the user to input it manually anywhere.

Reply:
Gotcha. Thus far the code hasn't made any assumptions on config file names just in case a user wants to test out different configurations, so that's why I thought this may have related to the config file name passed as an argument to run this script. I suggest sticking to the convention of specifying the config file name of interest in the vizualization config file for consistency's sake, plus not having to worry about what how we name the config files.